### PR TITLE
Update Docker artifacts for 4.1.0 GA release

### PR DESCRIPTION
**Pull request by IsuruMaduranga**
## Purpose

1. Doing necessary changes to support the APIM-4.1.0 GA release
2. Changing default commands provided in README files to map port 8243 (Passthrough or NIO HTTPS transport) to the docker host. This slight change will make life a bit easier for newcomers.
3. Following additions to the documentation

    - Instructions to build & run multi-architecture docker images ( Ubuntu & CentOS )
    - Instructions to build wso2am-alpine images on Apple Silicon
    - Instructions to run official wso2am images

### CODE REVIEW

File: dockerfiles/jdk8/ubuntu/apim/Dockerfile [hunk 1]
```diff
@@ -79,11 +79,11 @@ ARG USER_GROUP_ID=802
 ARG USER_HOME=/home/${USER}
 # build arguments for WSO2 product installation
 ARG WSO2_SERVER_NAME=wso2am
-ARG WSO2_SERVER_VERSION=4.1.0-beta
+ARG WSO2_SERVER_VERSION=4.1.0
 ARG WSO2_SERVER_REPOSITORY=product-apim
 ARG WSO2_SERVER=${WSO2_SERVER_NAME}-${WSO2_SERVER_VERSION}
 ARG WSO2_SERVER_HOME=${USER_HOME}/${WSO2_SERVER}
-ARG WSO2_SERVER_DIST_URL=https://github.com/wso2/${WSO2_SERVER_REPOSITORY}/releases/download/v${WSO2_SERVER_VERSION}/${WSO2_SERVER}.zip
+ARG WSO2_SERVER_DIST_URL=https://github.com/wso2/${WSO2_SERVER_REPOSITORY}/releases/download/v${WSO2_SERVER_VERSION}-rc2/${WSO2_SERVER}-rc2.zip
```
Comment (RakhithaRR) [inline dockerfiles/jdk8/ubuntu/apim/Dockerfile:86]: Let's remove the `-rc2` parts in this Dockerfile.

File: dockerfiles/alpine/apim/Dockerfile [hunk 2]
```diff
@@ -61,7 +61,7 @@ RUN echo Verifying install ... \
     && echo Complete.
 
 LABEL maintainer="WSO2 Docker Maintainers <dev@wso2.org>"  \
-      com.wso2.docker.source="https://github.com/wso2/docker-apim/releases/tag/v4.0.0.4"
+      com.wso2.docker.source="https://github.com/wso2/docker-apim/releases/tag/v4.1.0.0"
```
Comment (RakhithaRR) [inline dockerfiles/alpine/apim/Dockerfile:64]: Should be v4.1.0.1

Comment (IsuruMaduranga) [reply dockerfiles/alpine/apim/Dockerfile]: fixed

File: dockerfiles/jdk8/ubuntu/apim/Dockerfile (hunk 1 above)
Comment (IsuruMaduranga) [reply dockerfiles/jdk8/ubuntu/apim/Dockerfile]: fixed

**Review by RakhithaRR — APPROVED** [review]

**Review by RakhithaRR — APPROVED** [review]

**Pull request merged**